Tools/setup/Dockerfile

```diff
@@ -0,0 +1,55 @@
+#
+# PX4 base development environment
+#
+
+FROM ubuntu:20.04
+LABEL maintainer="Daniel Agar <daniel@agar.ca>, Ramon Roche <mrpollo@gmail.com>"
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV LANG C.UTF-8
+ENV LC_ALL C.UTF-8
+
+# Installing required utilities
+RUN apt-get update && apt-get -y --quiet --no-install-recommends install \
+  ca-certificates \
+  gnupg \
+  lsb-core \
+  lsb-release \
+  sudo \
+  software-properties-common \
+  wget \
+  gosu \
+  ;
+
+# Install PX4 Requirements
+COPY Tools/setup/requirements.txt /tmp/requirements.txt
+COPY Tools/setup/ubuntu.sh /tmp/ubuntu.sh
+# We support pre-downloading the gcc arm none eabi compiler
+# to speed up build times, if the file is not present when
+# building, the ubuntu.sh script will download it from source
+COPY *gcc-arm-none-eabi-9-2020-q2-update-linux.tar.bz2 /tmp/gcc-arm-none-eabi-9-2020-q2-update-linux.tar.bz2
+# The PATH env is set within ubuntu.sh, but given how we
+# are running the image using `gosu` to avoid read-only problems
+# with the filesystem the env variable does not persist
+ENV PATH="/opt/gcc-arm-none-eabi-9-2020-q2-update/bin:$PATH"
+ENV PATH="/opt/jdk-14.0.2+12/bin:$PATH"
+RUN bash /tmp/ubuntu.sh --from-docker --with-java --with-rtps --no-sim-tools
```

Comment:
Without the simulation tools, you can't run SITL using `docker_run.sh` in combination with this.  Maybe add an `ARG` to the Dockerfile to enable/disable the `--no-sim-tools` option?  Then we could use this Dockerfile to build either a one-container-to-rule-them image, or more minimalist images for Arm vs. SITL development.

A `build_docker.sh` script could also be a handy reference:
```
docker build -t px4io/px4-dev-arm:latest -f Tools/setup/Dockerfile . --build-arg SITL=no
docker build -t px4io/px4-dev-sitl:latest -f Tools/setup/Dockerfile . --build-arg ARM=no
docker build -t px4io/px4-dev-complete -f Tools/setup/Dockerfile .
```

Reply:
I like the idea of having a `docker_build.sh` interface, I will make it so we can directly pass any `ubuntu.sh` build arguments and will only hard code `--from-docker` directly on the Dockerfile since that is the only requirement.

Regarding the multiple images, the goal of this PR is to have one to rule them all for development and CI (including simulation).

The nice thing about the `build_docker.sh` interface you propose is that we can use that to build whatever we want w/o having to modify the Dockerfile directly.

Reply:
I just updated the PR to include the build script, which now supports any optional argument within `ubuntu.sh`, I also updated the main PR description with the new info.

Thanks for the suggestion.

Reply:
The original motivation here was to get down to 1 good container that serves the vast majority of what we actually need, so adding options and better scripts for handling those options kind of defeats the purpose.

Each actual Dockerfile needed should be static and describe a single thing.

Reply:
After a few days' break and returning to the PR review, I think it's best if the Dockerfiles are self contained instead of a parametrized script. The benefit of being able to trace down the exact build instructions heavily outweighs the ease of building a new image